explainer.md

```diff
@@ -215,6 +244,175 @@ third-party then signs either the exchanges in a package or the package as a
+as prefetches. If those point at signed exchanges, this process repeats.
+
+### No nested signed exchanges
+
+To limit the complexity of the implementation, we're currently planning to
+disallow signed exchanges that contain either signed exchanges or redirects.
+This may change if use cases come up or if the implementation turns out to be
+simpler than expected.
+
+### Caching the signed response
+
+If the signed exchange was requested as a navigation or subresource (i.e. *not*
+prefetches), the client tries to cache the request→response pair it contains.
+
+This *doesn't* happen if:
+
+* The signed exchange's request headers aren't sufficiently similar (TBD) to the
+  request headers the client would use for a normal request in the same context.
+  This prevents a malicious intermediate from sticking the wrong
+  content-negotiated resource in the HTTP cache.
+* There's a response in the HTTP (or any?) cache with a newer Date header
+  than the signed exchange's response. This prevents some downgrade attacks.
+
+In either of these cases, the client just skips to the redirect in the next
+step.
+
+If we're still here, the signed exchange is put into a "use-once" cache similar
+to the [preload cache](https://github.com/whatwg/fetch/issues/590) and, if the
+response headers allow it (and review of this explainer indicates it's a good
+idea), the [HTTP cache](https://tools.ietf.org/html/rfc7234).
+
+If we put the signed exchange in the HTTP cache, its freshness has to be bounded
+by the shorter of the normal HTTP cache lifetime or the signature's expiration.
+This is more strict than the bound on cache freshness when it crosses a
+certificate or OCSP response expiration in order to partially mitigate
+[downgrade
+attacks](https://wicg.github.io/webpackage/draft-yasskin-http-origin-signed-responses.html#seccons-downgrades).
+For *later* loads of the logical URL (in particular, not the load that's
+happening through the signed exchange, since it's fulfilled using the
+above-mentioned "use-once" cache), a stale entry can be revalidated in the
+following ways:
+
+* If the `Signature` is expired but the HTTP caching information is fresh, the
+  client can fetch the `validityUrl `to update just the signature. It *must not*
+  send an `If-None-Match` or `If-Modified-Since` request to update the cache,
+  because the `Signature` expiring means we don't trust the claimed ETag or date
+  anymore.
+* If the `Signature` is valid but the HTTP caching information is stale, the
+  client can send the logical URL an `If-None-Match` or `If-Modified-Since`
+  request hoping for a 304. Note that the client has to keep the original
+  response headers if it intends to use the `validityUrl` to update the
+  signature in the future.
+* If neither is valid, the client could first update the signature and then
+  check for a 304 (or even do both concurrently), but it may be easier to just
+  do an unconditional request.
```

Comment:
I feel like I'm missing something here -- what does "first update the signature and then check for a 304" mean?

Reply:
To update the signature, the client follows https://wicg.github.io/webpackage/draft-yasskin-http-origin-signed-responses.html#rfc.section.3.6. Then "check for a 304" is probably equivalent to "makes a conditional request". It's intended to be the two bullets above.